pprint/src-0/TPrintImpl.scala

```diff
@@ -0,0 +1,104 @@
+package pprint
+
+import language.experimental.macros
+import scala.reflect.macros.TypecheckException
+
+trait TPrintLowPri{
+  inline given default[T] as TPrint[T] = ${ TPrintLowPri.typePrintImpl[T] }
+}
+
+object TPrintLowPri{
+
+  import scala.quoted._
+  import sourcecode.Text.generate
+
+  def typePrintImpl[T](using qctx: QuoteContext, t: Type[T]): Expr[TPrint[T]] = {
+
+    import qctx.tasty._
+    import util._
+
+    def literalColor(cfg: Expr[TPrintColors], s: Expr[fansi.Str]) = {
+      '{ $cfg.typeColor($s).render }
+    }
+
+    def printSymString(cfg: Expr[TPrintColors], s: String) =
+      if (s.toString.startsWith("_$")) "_"
+      else s.toString.stripSuffix(".type")
+
+    def printBounds(cfg: Expr[TPrintColors])(lo: Type, hi: Type) = {
+      val loTree =
+        if (lo =:= typeOf[Nothing]) None else Some('{ " >: " + ${ rec0(cfg)(lo) }})
+      val hiTree =
+        if (hi =:= typeOf[Any]) None else Some('{ " <: " + ${ rec0(cfg)(hi) }})
+      loTree.orElse(hiTree).map((expr) => '{ "_" + $expr }).getOrElse('{ "_" })
+    }
+
+    def printSym(cfg: Expr[TPrintColors], s: String): Expr[String] = {
+      val expr = Expr(s)
+      literalColor(cfg, '{ fansi.Str($expr) })
+    }
+
+    //TODO: We don't currently use this method
+    def prefixFor(cfg: Expr[TPrintColors])(pre: TypeTree, sym: String): Expr[String] = {
+      // Depending on what the prefix is, you may use `#`, `.`
+      // or even need to wrap the prefix in parentheses
+      val sep = pre match {
+        case x if x.toString.endsWith(".type") =>
+          '{ ${rec0(cfg)(pre.tpe)} + "." }
+      }
+      '{ $sep + ${ printSym(cfg, sym) } }
+    }
+
+
+    def printArgs(cfg: Expr[TPrintColors])(args: List[TypeOrBounds]): Expr[String] = {
+      val added = args.map {
+        case tpe: Type =>
+          rec0(cfg)(tpe, false)
+        case TypeBounds(lo, hi) =>
+          printBounds(cfg)(lo, hi)
+      }.reduceLeft { (l, r) =>
+        '{ $l  + ", " + $r }
```

Comment:
We seem to only need string additions from the quotes. It would be good to abstract this and have a single quote with this operation in the code. This will reduce the generated code size and make the code simpler to read.
One way to do his would be

```scala
extension (expr: Expr[String])
  def +(other: Expr[String])(using QuoteContext): Expr[String] =
    '{ $expr + $other }
```

```suggestion
        l + Expr(", ") + r
```

Reply:
Or with the `StringBuffer` idea bellow
```scala
extension (expr: Expr[StringBuffer])
  def appended(sb: Expr[StringBuffer], str: Expr[String]): Expr[StringBuffer] =
    '{ $sb.append($str) }
```

```suggestion
  sb.appended(l).appended(Expr(", ")).appended(r)
```